### DISCUSSION TITLE
FreeBSD alias plugin similar to archlinux, macos, etc.

### DISCUSSION THREAD

**Discussion by RanceH** [ohmyzsh/ohmyzsh#11947] [Ideas]
### If the feature solves a problem you have, specify it here.

I'm an avid FreeBSD user on servers, and just switched to zsh and omz from a failed attempt at using the fish shell with the similarly titled omf.

On my linux boxes there are already plugins in omz for my distros, but I noticed no such plugin exists for FreeBSD.

### Describe the proposed feature.

Addition of a plugin supporting common system admin aliases of FreeBSD similar to archlinux or macos plugins already available.

**Comment by RanceH** [comment]
I'm willing to fork the project and create a first version of the plugin, but wanted to ask for the discussion of usefulness to the greater community before doing all that.

**Reply by robbyrussell** [reply]
I think the most important to ask yourself is, what would be useful for you?

If you end up making a custom plugin and think it'd be useful for others, you should definitely submit a PR!

**Comment by okapia** [comment]
If you've got something that _you_ find useful, open a PR.

I'm a FreeBSD user but really can't find much in my configuration that is especially FreeBSD specific. Stuff like `freebsd-update fetch install` could perhaps be aliased.

From a quick glance at the archlinux plugin, it looks mostly like pacman aliases, often prefixing them with `sudo`. Given that FreeBSD doesn't include `sudo` in base and many FreeBSD people either use `doas` instead or prefer plain `su` I'm not sure how useful that would be. Note also that `pkg` supports it's own aliases - see the ALIAS block in `/usr/local/etc/pkg.conf`. The macos plugin seems to be functions reflecting areas where macos differs from other Unix systems.

In my opinion, even for ArchLinux, it should test `${+commands[sudo]}` and `(( EUID ))` before including `sudo` in the alias.

**Reply by pepoluan** [reply]
I submitted an issue specifically about enabling use of alternative subexecutor such as `doas`: https://github.com/ohmyzsh/ohmyzsh/issues/12245

**Reply by pepoluan** [reply]
Created PR #12261 implementing ideas in #12245 .  
The hope is that plugins that only need a subexecutor rather than specifically needing sudo, can now just use `subex` and it will match whatever subexecutor is being used, be it `doas` (that I also use on Gentoo Linux) or `sudo`.
